lib/utils/utf8.c

```diff
@@ -79,3 +80,37 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const char *str, size_t maxlen)
+{
+	int i = 0, nbyte = 0;
+	size_t len = strlen(str);
+	unsigned char *buf = (unsigned char *)str;
+
+	/* Return false when the string is greater than the maximum length. */
+	if (len > maxlen) {
+		return false;
+	}
+
+	while (i < len) {
+		if (buf[i] <= 0x7F && buf[i] >= 0x00) {
+			i++;
+			continue;
+		} else {
+			if (buf[i] <= 0xDF && buf[i] >= 0xC2) {
+				nbyte = 2;
+			} else if (buf[i] <= 0xEF && buf[i] >= 0xE0) {
+				nbyte = 3;
+			} else if (buf[i] <= 0xF4 && buf[i] >= 0xF0) {
```

Comment:
```suggestion
			if (buf[i] < SEQUENCE_LEN_3_BYTE && buf[i] >= 0xC2) {
				nbyte = 2;
			} else if (buf[i] < SEQUENCE_LEN_4_BYTE && buf[i] >= SEQUENCE_LEN_3_BYTE) {
				nbyte = 3;
			} else if (buf[i] <= 0xF4 && buf[i] >= SEQUENCE_LEN_4_BYTE) {
```

We don't have defined values for the remaining values, `0xC2` and `0xF4`, but I suggest to define these and use them, to avoid too many magical and undocumented values. 

You can probably also the remove the `&& <check>` for these if the `buf[i] >= 0xC2` is applied first, as the preceding checks has already verified the value range. This would be a easy optimization

Reply:
That's fine 👍